app/helpers/layouts_helper.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module LayoutsHelper
+  def parent_layout(layout)
```

Comment:
My knowledge here is admittedly a little rusty.  But I wonder we could avoid direct access to undocumented actionview internals like this - maybe something like https://lortza.github.io/2020/08/22/nested-layouts.html. Would require a mod to `single_column` layout to check for custom content rendered by the exchanges_base layout. If I'm following correctly, that might be another way to get to what you want, and uses more well-documented public APIs.

Reply:
I like it. Good idea. Thank you